libleptongui/src/gschem_options.c

```diff
@@ -129,12 +129,12 @@ gschem_options_cycle_net_rubber_band_mode (GschemOptions *options)
 void
 gschem_options_cycle_snap_mode (GschemOptions *options)
 {
-  SNAP_STATE next_snap_mode;
+  SchematicSnapMode next_snap_mode;
 
   g_return_if_fail (options != NULL);
 
   /* toggle to the next snap state */
-  next_snap_mode = (SNAP_STATE) ((options->snap_mode + 1) % SNAP_STATE_COUNT);
+  next_snap_mode = (SchematicSnapMode) ((options->snap_mode + 1) % SNAP_MODE_COUNT);
```

Comment:
It's a minor thing, but I would rewrite it in more human-friendly manner, something like:
```
SchematicSnapMode mode = (SchematicSnapMode) (options->snap_mode + 1);
next_snap_mode = (SchematicSnapMode) (mode == SNAP_MODE_COUNT ? 0 : mode);
```
IMHO, it more clearly expresses the real purpose of this code.
The same for `gschem_options_cycle_grid_mode()` above.

Reply:
Well, AFAIK, there is no standard defining that `enum` values have to be sequential integers starting with one, so both the existing and your code are wrong trying to rely upon this.  While most of the modern C compilers do (as it seems) TRT with our code, in order to properly manage these values, we have to explicitly assign values to each `enum` value, IMO.

Reply:
Anyways, please go for it, if you wish so.